test/ui.spec.ts

```diff
@@ -94,279 +146,1444 @@ describe('Signer integration', function () {
+          'AABAgAAABNMaXN0IHNpemUgZXhjZWVkcyA1CQEAAAAFJGYwXzIAAAACCQEAAAAFJGYwXzEAAAACCQEAAAAFJGYwXzEAAAACCQEAAAAFJGYwXzEAAAACCQEAAAAFJGYwXzEAAAACCQEAAAAFJGYwXzEAAAA' +
+          'CBQAAAAUkYWNjMAAAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAgAAAAAAAAAAAwAAAAAAAAAABAAAAAAAAAAABQAAAABWejDo',
+      };
+
+      await performSetScriptTransaction.call(this, data);
+
+      await approveMessage.call(this);
+      await closeMessage.call(this);
+
+      const result = (await getApproveResult.call(this)) as [
+        BroadcastedTx<SignedTx<SignerSetScriptTx>>
+      ];
+
+      const [parsedApproveResult] = result;
+      const expectedApproveResult = {
+        type: 13 as const,
+        version: 2,
+        senderPublicKey: user1.publicKey,
+        script: data.script,
+        fee: 300000,
+        chainId,
+      };
+
+      const bytes = makeTxBytes({
+        ...expectedApproveResult,
+        timestamp: parsedApproveResult.timestamp,
+      });
+
+      expect(parsedApproveResult).to.deep.contain(expectedApproveResult);
+      expect(parsedApproveResult.id).to.equal(base58Encode(blake2b(bytes)));
+
+      expect(
+        verifySignature(user1.publicKey, bytes, parsedApproveResult.proofs[0])
+      ).to.be.true;
     });
 
-    it('with no args but single payment', async function () {
-      await signedTxShouldBeValid.call(
-        this,
-        'invoke',
-        INVOKE_NO_ARGS_SINGLE_PAYMENTS,
-        By.xpath("//div[contains(@class, 'scriptInvocation-transaction')]")
-      );
+    async function performInvokeTransaction(
+      this: mocha.Context,
+      data: InvokeArgs
+    ) {
+      const { waitForNewWindows } = await Windows.captureNewWindows.call(this);
+      await this.driver.executeScript(data => {
+        window.result = window.signer
+          .invoke(data)
+          .broadcast({ confirmations: 1 });
+      }, data);
+
+      [messageWindow] = await waitForNewWindows(1);
+      await this.driver.switchTo().window(messageWindow);
+      await this.driver.navigate().refresh();
+    }
+
+    it('Invoke script with payment', async function () {
+      await changeKeeperAccountAndClose.call(this, 'issuer');
+      await waitKeeperAccountChanged.call(this, issuer);
+
+      const data: InvokeArgs = {
+        dApp: user1.address,
+        call: {
+          function: 'deposit',
+          args: [],
+        },
+        payment: [{ assetId: null, amount: 200000000 }],
+      };
+
+      await performInvokeTransaction.call(this, data);
+
+      await approveMessage.call(this);
+      await closeMessage.call(this);
+
+      const result = (await getApproveResult.call(this)) as [
+        BroadcastedTx<SignedTx<SignerInvokeTx>>
+      ];
+
+      const [parsedApproveResult] = result;
+      const expectedApproveResult = {
+        type: 16 as const,
+        version: 2,
+        senderPublicKey: issuer.publicKey,
+        dApp: data.dApp,
+        call: data.call,
+        payment: data.payment as NonNullable<typeof data.payment>,
```

Comment:
Seems like `!` should work here to. And there are more similar casts in this file